### ISSUE TITLE
Probe should iterate the list of model statuses when model version not specified

### ISSUE THREAD

**Issue by adriangay** [codycollier/tfs-model-status-probe#1]
For a given model name, we control which model version Tensorflow Serving (TFS) serves by dynamically updating the `model_config_list` object which is being polled by TFS. Models are retrained frequently and on every successful training iteration, automation updates the `model_config_list` object.

The Tensorflow Serving ModelStatusResponse does not necessarily return the `AVAILABLE` model as the first item. There are at least two cases where this is not so:
- via `model_config_list` updates, the model base path varies and the model version is not continuously sequentially increasing
- via  `model_config_list` updates, model rollback to a previous model version means the latest version becomes `ENDED`

We want to use your probe in place of some custom code, but have encountered this issue. When calling the probe we see that the probe assumes the `AVAILABLE` model will be first in the list returned:

```
$ /tfs_model_status_probe/tfs_model_status_probe_1.0.1_linux_amd64 -addr=:8500 -model-name=ncr
2022/06/27 11:12:31 ModelStatusResponse: model_version_status:{version:41823  state:END  status:{}}  model_version_status:{version:41822  state:END  status:{}}  model_version_status:{version:374  state:AVAILABLE  status:{}}
2022/06/27 11:12:31 Servable state is END
```
We can see that `374` state is `AVAILABLE`, but the probe did not see this. Looking into the probe source code we find that if a model version is not passed to the probe, ie. just a model name, it takes the first status in the list; it does not iterate the list to find an `AVAILABLE` status when a specific model version is not specified;

https://github.com/codycollier/tfs-model-status-probe/blob/3a4e7cfa188035178ebc3253707e0d1685937009/main.go#L68

Because we control which model TFS serves using `model_config_list`, we don't pass `version` in the `PredictionService` `PredictRequest` `ModelSpec` which means the service calling TFS does not need to know model version, it just uses whatever is `AVAILABLE`.

We confirm in the TFS source code that it returns model statuses in whatever order they are stored in the structure used by `monitor.GetVersionStates(model_name)`:

https://github.com/tensorflow/serving/blob/04b8676730a882cab61b7772097f2c23c0447ef9/tensorflow_serving/model_servers/get_model_status_impl.cc#L95

**Comment by codycollier** [comment]
Thanks for the submission and insight @adriangay!  This makes sense to me.  I'll work up a change ~in the next week or so (or feel free to submit a PR if you have one)~.

**Comment by codycollier** [comment]
A PR is up, which I think captures the behavior you've described @adriangay.  Feel free to take a look and comment.

**Comment by alexgeorgousis** [comment]
Thanks for addressing the issue so quickly @codycollier 🙏

**Comment by adriangay** [comment]
🙇 thank you for fixing this

**Comment by codycollier** [comment]
Thanks everyone! PR was merged and a new release with platform binaries is [available in v1.0.2](https://github.com/codycollier/tfs-model-status-probe/releases/tag/v1.0.2)
